pkg/nvlib/info/builder.go

```diff
@@ -0,0 +1,39 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package info
+
+type builder struct {
+	root root
+}
+
+// New creates a new instance of the 'info' Interface.
+func New(opts ...Option) Interface {
+	b := &builder{}
+	for _, opt := range opts {
+		opt(b)
+	}
+	if b.root == "" {
+		b.root = "/"
+	}
+	return b.build()
+}
+
+func (b *builder) build() Interface {
+	return &infolib{
+		root: root(b.root),
+	}
+}
```

Comment:
Is this extra function necessary? Do we call it anywhere other than in in `New()`?

Reply:
No, your're right. The initial implementation was structured a bit differently, and I did not change this when I simplified things. Let me rework it.